### PR TITLE
RMA episodes

### PR DESCRIPTION
**Pull request by robertodr**
I have been using diagrams.net to draw the figures. The files with all drawings is `content/diagrams/intermediate-mpi.drawio`. You can edit online and directly save to the repository.

There are a lot of commits (because I've been saving frequently the drawn diagrams) I will squash the commits once I'm done with these two episodes.

### CODE REVIEW

**Comment by wikfeldt** [comment]
high quality stuff. I really like the generality offered by the substitutions, but it must add some cognitive overload when writing it!

File: content/collective-communication-pt1.rst [hunk 1]
```diff
@@ -21,6 +21,22 @@ Topic introduction here
 You really want to browse this page alongside the source of it, to see
 how this is implemented.  See the links at the to right of the page.
 
+.. table:: Point-to-point communication functions
```
Comment (mabraham) [inline content/collective-communication-pt1.rst]: This might not be in the right file

File: content/custom.py [hunk 2]
```diff
@@ -50,6 +52,21 @@
 ]
 # collectives
```
Comment (mabraham) [inline content/custom.py:57]: same, these are point-to-point

File: content/collective-communication-pt1.rst (hunk 1 above)
Comment (robertodr) [reply content/collective-communication-pt1.rst]: Indeed. I was making a summary for myself and experimenting with tables. I'll stash it in some other file.

**Review by wikfeldt — COMMENTED** [review]
So much to like here! Great balance of text, figures, bullets, disucssion and function signatures. Nice usage of the emojis too, but we should make sure that participants know what they mean (table on index page?). I also wonder about the colors, could the contrasts be too weak for people with reduced eyesight or color blindness? And final feedback: are you planning any coding exercises?

**Comment by robertodr** [comment]
- Yep, exercises are coming :) I wanted to write up the other "easier" episodes (comms&groups, datatypes) They will be in clean-up PR later.
- The colors come from here: https://personal.sron.nl/~pault but I adjusted the opacity. 
- Good idea with the explanation table.

**Comment by robertodr** [comment]
It would be great if the signature directive could extract the signature from `mpi.h` directly for us :smile_cat:

**Review by wikfeldt — APPROVED** [review]
reading automatically from mpi.h would indeed be cool! 
Anyways, this looks very good so merging

**Pull request merged**